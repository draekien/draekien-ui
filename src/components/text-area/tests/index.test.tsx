/** @jsx jsx */
import { jsx } from 'theme-ui';
import { render } from '@testing-library/react';
import { TextArea } from '..';

describe('TextArea component', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <TextArea
        id="area"
        label="textarea"
        helpText="helpText"
        placeholder="placeholder"
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('should not render label and helpText when not provided', () => {
    const { container } = render(<TextArea id="area" />);

    expect(container.querySelector('label')).toBe(null);
  });
});
