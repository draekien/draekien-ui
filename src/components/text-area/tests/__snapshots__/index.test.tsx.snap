// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea component should match snapshot 1`] = `
<div>
  <div
    class="css-3j7u2z-TextArea"
  >
    <div
      class="css-1yiveqa-InputWrapper"
    >
      <label
        class="css-1x6zbg6-InputWrapper"
        for="area"
      >
        <div
          class="css-1x18leo-Text"
        >
          textarea
        </div>
      </label>
      <textarea
        class="css-ki8o5-TextArea"
        id="area"
        placeholder="placeholder"
        rows="1"
        style="height: 6.5rem; min-height: 6.5rem; max-height: 6.5rem; overflow-x: hidden; word-wrap: break-word;"
      />
      <label
        class="css-1x6zbg6-InputWrapper"
        for="area"
      >
        <div
          class="css-1x18leo-Text"
        >
          helpText
        </div>
      </label>
    </div>
  </div>
</div>
`;
